### ISSUE TITLE
Regenerating certificates for amazonec2 host fails

### ISSUE THREAD

**Issue by beatlevic** [docker/machine#2320]
Running Ubuntu 15.04 with docker 1.9 and docker-machine 0.5.1:

1. Created machine on EC2: docker-machine create -d amazonec2 whatever
2. deploy and run a container. Everything works fine.
3. Stopped EC2 instance.
4. 2 weeks later starting the EC2 instance again and machine has a new IP
5. eval "$(docker-machine env whatever)"
Error running connection boilerplate: Error checking and/or regenerating the certs: There was an error validating certificates for host "xx.xx.xx.xxx:2376": tls: DialWithDialer timed out
You can attempt to regenerate them using 'docker-machine regenerate-certs name'.
Be advised that this will trigger a Docker daemon restart which will stop running containers.

6. docker-machine regenerate-certs whatever
Just hangs forever after "Regenerating TLS certificates"

This issue is similar to what people experienced with docker/toolbox: https://github.com/docker/toolbox/issues/275

**Comment by nathanleclaire** [comment]
If you run `regenerate-certs` it should fix the certificate having an improper `altName`.  Can you provide the output of `docker-machine --debug regenerate-certs instance`?

**Comment by beatlevic** [comment]
`docker-machine --debug regenerate-certs instance` would hang forever:

(instance) Calling RPCServerDriver.GetSSHPort
(instance) Calling RPCServerDriver.GetSSHKeyPath
(instance) Calling RPCServerDriver.GetSSHUsername
Using SSH client type: external
About to run SSH command:
sudo docker version

So I checked to see if docker was running correctly on my machine and used `docker-machine ssh instance`. Turns out is wasn't and I found that it was depending on my consul server (multi-host-networking), which also got a new IP address. These were my `/etc/default/docker` arguments:

DOCKER_OPTS='
-H tcp://0.0.0.0:2376
-H unix:///var/run/docker.sock
--storage-driver aufs
--tlsverify
--tlscacert /etc/docker/ca.pem
--tlscert /etc/docker/server.pem
--tlskey /etc/docker/server-key.pem
--label provider=amazonec2
--cluster-store=consul://x.x.x.x:8500
--cluster-advertise=eth0:0

Updating the --cluster-store to the correct consul IP address and removing `/var/lib/docker/network/files/local-kv.db` (https://github.com/docker/docker/issues/17846), I could get docker running again 'sudo service docker restart'.

In other words if your consul server is unreachable and/or changed its IP address, then docker will fail to start. On AWS I'm going to assign an elastic IP to my consul server.

**Comment by NaitikShah** [comment]
Steps which started my machine normally:

- Rebooting my ec2 manager node from AWS management console.
- `docker-machine regenerate-certs -f [your-machine-name]`

By doing this steps I was able to ssh my manager node in `docker-swarm` and also all my services were auto-restarted moreover the IP of the machines had not changed after regenerating certs.

**Comment by stall84** [comment]
@NaitikShah Good moves. I tried this process. Just for others that might want to try it this was the full process: 
1. Stop the aws-machine on your local/dev.
2. Log in to AWS (web) Console. START the Instance from the AWS Console, then Reboot it from the 'Instance State' button/tab.
3. After it reboots and it up from AWS Console. Go back to your local machine's terminal and run `docker-machine regenerate-certs -f [your-machine-name]`
4. It should not hang up and re-load docker and you should then be able to `> eval $(docker-machine [your-machine-name] )`
